src/distance/edge-filters.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef SRC_DISTANCE_EDGE_FILTERS_HPP_
+#define SRC_DISTANCE_EDGE_FILTERS_HPP_
+
+#include <memory>
+#include <optional>
+
+#include "command-line/parsing/options.hpp"
+#include "common/pipeline/pipelines.hpp"
+#include "distance/edge.hpp"
+
+namespace found {
+
+
+using EdgeFilteringAlgorithms = ModifyingPipeline<Points>;
+
+/**
+ * Abstract superclass for edge-filtering algorithms.
+ * modifies Points in-place (ModifyingStage<Points>).
+ */
+class EdgeFilteringAlgorithm : public ModifyingStage<Points> {
+ public:
+    EdgeFilteringAlgorithm() = default;
+    virtual ~EdgeFilteringAlgorithm() = default;
+    // Implementations override void Run(Points &)
+};
+
+/**
+ * A trivial implementation to ensure linkage / prevent undefined references.
+ * Does nothing to the Points (no-op filter).
+ */
+class NoOpEdgeFilter : public EdgeFilteringAlgorithm {
+ public:
+    NoOpEdgeFilter() = default;
+    ~NoOpEdgeFilter() override = default;
+
+    void Run(Points &pts) override {
+        // intentionally no-op
+        (void)pts;
+    }
+};
+
+namespace detail {
+inline NoOpEdgeFilter kNoOpEdgeFilter{};
+}  // namespace detail
+
```

Comment:
Neither of the ProvideEdgeFilterAlgorithm functions should exist here. They should be in `stage-providers`.